firefox-ios/Client/Frontend/Home/HomepageViewController.swift

```diff
@@ -518,6 +518,7 @@ extension HomepageViewController: UICollectionViewDelegate, UICollectionViewData
     }
 
     func numberOfSections(in collectionView: UICollectionView) -> Int {
+        if viewModel.shouldReloadView { reloadView() }
```

Comment:
🤔 It can't be this simple, but I also can't think of a reason this won't work, nice one!
We should uplift this to 127 so we can get beta feedback for it. 🤞